src/app/pages/dashboard/affected.people/components/affected.people.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, Output, EventEmitter } from '@angular/core';
+import { Headers } from '@angular/http';
+import { Observable } from 'rxjs/Rx';
+
+import { InvolvedPartyModel } from '../../InvolvedParty';
+import { AffectedModel } from '../../affecteds';
+import { AffectedPeopleToView, AffectedPeopleModel } from './affected.people.model';
+import {
+    ResponseModel, DataService,
+    DataServiceFactory, DataProcessingService,
+    IServiceInretface, UtilityService
+} from '../../../../shared';
+
+@Injectable()
+export class AffectedPeopleService {
+    private _dataService: DataService<InvolvedPartyModel>;
+    private _bulkDataService: DataService<AffectedPeopleModel>;
+    constructor(private dataServiceFactory: DataServiceFactory) {
+        let option: DataProcessingService = new DataProcessingService();
+        this._dataService = this.dataServiceFactory
+            .CreateServiceWithOptions<InvolvedPartyModel>('InvolvedParties', option);
+        this._bulkDataService = this.dataServiceFactory
+            .CreateServiceWithOptions<AffectedPeopleModel>('AffectedPersonBatch', option);
+    }
+
+    GetAll(): Observable<ResponseModel<InvolvedPartyModel>> {
+        return this._dataService.Query()
+            .Expand('Affecteds($expand=AffectedPeople($expand=Passenger))')
+            .Execute();
+    }
+    GetFilterByIncidentId(): Observable<ResponseModel<InvolvedPartyModel>> {
+        return this._dataService.Query()
+            .Filter('IncidentId eq ' + '88')
+            .Expand('Affecteds($expand=AffectedPeople($expand=Passenger,Crew))')
+            .Execute();
+    }
+
+    FlattenData(involvedParty: InvolvedPartyModel): any {
+        let affectedPeopleForView: AffectedPeopleToView[];
+        let affectedPeople: AffectedPeopleModel[];
+        let affected: AffectedModel;
+        affected = UtilityService.pluck(involvedParty, ['Affecteds'])[0][0];
+        affectedPeople = UtilityService.pluck(affected, ['AffectedPeople'])[0];
+        affectedPeopleForView = affectedPeople.map(function (dataItem) {
+            let item = new AffectedPeopleToView();
+            item.AffectedId = dataItem.AffectedId;
+            item.AffectedPersonId = dataItem.AffectedPersonId,
+                item.PassengerName = dataItem.Passenger != null ? dataItem.Passenger.PassengerName : '';
```

Comment:
Add a function in the UtilityService to check "null" or undefined and put some default value